cartography/graph/statement.py

```diff
@@ -59,30 +64,31 @@ def as_dict(self):
             "iterationsize": self.iterationsize,
         }
 
-    def _run(self, session):
+    def _run(self, tx: neo4j.Transaction) -> neo4j.StatementResult:
         """
         Non-iterative statement execution.
         """
-        return session.run(self.query, self.parameters)
+        return tx.run(self.query, self.parameters)
 
-    def _run_iterative(self, session):
+    def _run_iterative(self, tx: neo4j.Transaction) -> None:
         """
         Iterative statement execution.
 
         Expects the query to return the total number of records updated.
         """
         self.parameters["LIMIT_SIZE"] = self.iterationsize
 
-        # TODO improve this
-        done = False
-        while not done:
-            results = self._run(session)
-            done = False
-            for r in results:
-                if int(r['TotalCompleted']) == 0:
-                    done = True
+        total_completed = -1
```

Comment:
might be cleaner here to just break when the total completed is 0, ie:
```
        while True:
            result: neo4j.StatementResult = self._run(tx)
            record: neo4j.Record = result.single()
            # TODO: use the BoltStatementResultSummary object to determine the number of items processed
            total_completed = int(record['TotalCompleted'])
            logger.debug("Processed %d items", total_completed)

            # Ensure network buffers are cleared
            result.consume()
            if total_completed == 0:
                break
```

Reply:
I guess it's a style thing, but seeing code like `while True` scares me. I don't have a strong opinion either way though.